### PR TITLE
[MRG]: Add support for reading curry EEG files

### LINKED ISSUE #1070: Add support for Curry .cdt input files
### Describe the problem

It seems, bids conversion does not support Curry files (.cdt). Is there any reason for this and would anything speak against adding support for it?

### Describe your solution

One could extend the `ALLOWED_INPUT_EXTENSIONS` in `mne_bids.confg`.

### Describe possible alternatives

The alternative is an additional conversion step into the supported file-formats (not practical).

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dengemann**
PR Description
--------------

This adds support for curry eeg files as input, closes #1070.
On local machine, not all tests pass on main branch, hence, I am not sure what certain test results mean for curry branch.
Let's see what CI's will say.

Merge checklist
---------------

Maintainer, please confirm the following before merging.
If applicable:

- [x] All comments are resolved
- [ ] This is not your own PR
- [x] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [x] New contributors have been added to [CITATION.cff](https://github.com/mne-tools/mne-bids/blob/main/CITATION.cff)
- [x ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

**Comment by hoechenberger** [comment]
Merging main just to be safe 😇

**Comment by hoechenberger** [comment]
@dengemann it appears the .dat extension is already in use for another reader.. maybe let's restrict ourselves to .cdt here?

**Comment by dengemann** [comment]
also we did not updated whats new etc.

**Comment by hoechenberger** [comment]
@sappelhoff We have a one-to-many mapping here, `.dat` can mean multiple file formats. What should we do?

**Comment by dengemann** [comment]
... perhaps we don't support it for now and wait till someone asks for it. I am personally OK with .cdt support.

**Comment by agramfort** [comment]
+1 for .cdt

**Comment by hoechenberger** [comment]
+1, let's only do `.cdt` for now then!!

**Comment by dengemann** [comment]
shall I open the PR again / continue or will you take over @hoechenberger  ?

**Comment by hoechenberger** [comment]
Please do work on this one as far / long as you can, @dengemann :) I don't have much time currently

**Comment by larsoner** [comment]
https://github.com/mne-tools/mne-bids/pull/1073 pushes commits that `xfail` the unrelated tests so that hopefully the feedback here will be more meaningful. You might want to `git fetch upstream && git merge upstream/main` after it's merged so that you can benefit from the dataset caching, too

**Comment by hoechenberger** [comment]
Pulled in `main` so we can see what's still failing here, as suggested by @larsoner

**Comment by agramfort** [comment]
@dengemann can you look into CI failures?

**Comment by dengemann** [comment]
Alright, made some progress here. CIs happy except for coverage.

**Comment by dengemann** [comment]
Ok it semes all is fine now.

File: doc/whats_new.rst [hunk 1]
```diff
@@ -24,7 +24,7 @@ Version 0.11 (unreleased)
 
 The following authors contributed for the first time. Thank you so much! 🤩
 
-* ...
+* `Denis Engemann`_
```
Comment (agramfort) [inline doc/whats_new.rst:27]: @dengemann you need to update authors.rst too

File: mne_bids/config.py [hunk 2]
```diff
@@ -59,7 +59,8 @@
                      '.lay': 'Persyst', '.dat': 'Persyst',
                      '.EEG': 'Nihon Kohden',
                      '.cnt': 'Neuroscan', '.CNT': 'Neuroscan',
-                     '.bin': 'EGI'}
+                     '.bin': 'EGI',
+                     '.cdt': 'n/a'}
```
[the remark this answers is not in the document]
Comment (dengemann) [reply mne_bids/config.py:63]: in my understanding Curry is used by different amps / manufacturers

[the remark this answers is not in the document]
Comment (dengemann) [reply mne_bids/config.py:63]: ah sorry. got confused. yes we could do "curry" for the fileformat

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
🙏 @dengemann

**Pull request merged**